test/contrib/emscripten/test_emscripten.py

```diff
@@ -0,0 +1,381 @@
+        await micropip.install('http://{testserver_http.http_host}:{testserver_http.http_port}/wheel/urllib3-2.0.7-py3-none-typing.Any.whl',deps=False)
+        import urllib3.contrib.emscripten.fetch
+        await urllib3.contrib.emscripten.fetch.wait_for_streaming_ready()
+        from urllib3.exceptions import TimeoutError
+        from urllib3.connection import HTTPConnection
+        conn = HTTPConnection("{testserver_http.http_host}", {testserver_http.http_port},timeout=1.0)
+        method = "GET"
+        url = "http://{testserver_http.http_host}:{testserver_http.http_port}/slow"
+        result=-1
+        try:
+            conn.request(method, url)
+            _response = conn.getresponse()
+            result=-3
+        except TimeoutError as e:
+            result=1 # we've got the correct exception
+        except BaseException as e:
+            result=-2
+        result
+"""
+    result = run_from_server.run_webworker(worker_code)
+    # result == 1 = success, -2 = wrong exception, -3 = no exception thrown
+    assert result == 1
+
+
+@copy_files_to_pyodide(file_list=[("dist/*.whl", "/tmp")], install_wheels=True)  # type: ignore[misc]
+def test_index_https(selenium: typing.Any, testserver_http: PyodideServerInfo) -> None:
+    @run_in_pyodide  # type: ignore[misc]
+    def pyodide_test(selenium, host: str, port: int) -> None:  # type: ignore[no-untyped-def]
+        from urllib3.connection import HTTPSConnection
+        from urllib3.response import HTTPResponse
+
+        conn = HTTPSConnection(host, port)
+        method = "GET"
+        path = "/"
+        url = f"https://{host}:{port}{path}"
+        conn.request(method, url)
+        response = conn.getresponse()
+        assert isinstance(response, HTTPResponse)
+        data = response.data
+        assert data.decode("utf-8") == "Dummy server!"
+
+    pyodide_test(selenium, testserver_http.http_host, testserver_http.https_port)
+
+
+@copy_files_to_pyodide(file_list=[("dist/*.whl", "/tmp")], install_wheels=True)  # type: ignore[misc]
+def test_non_streaming_no_fallback_warning(
+    selenium: typing.Any, testserver_http: PyodideServerInfo
+) -> None:
+    @run_in_pyodide  # type: ignore[misc]
+    def pyodide_test(selenium, host: str, port: int) -> None:  # type: ignore[no-untyped-def]
+        import urllib3.contrib.emscripten.fetch
+        from urllib3.connection import HTTPSConnection
+        from urllib3.response import HTTPResponse
+
+        conn = HTTPSConnection(host, port)
+        method = "GET"
+        path = "/"
+        url = f"https://{host}:{port}{path}"
+        conn.request(method, url, preload_content=True)
+        response = conn.getresponse()
+        assert isinstance(response, HTTPResponse)
+        data = response.data
+        assert data.decode("utf-8") == "Dummy server!"
+        # no console warnings because we didn't ask it to stream the response
+        assert not urllib3.contrib.emscripten.fetch._SHOWN_STREAMING_WARNING
```

Comment:
Is there a way to capture and assert on `console.log` calls from pyodide?

Reply:
yes. Can override it. It's a pain because it has to go inside each test for boring reasons, but I've put it in.